backends/nats/helpers.go

```diff
@@ -0,0 +1,75 @@
+package nats
+
+import (
+	"crypto/rand"
+	"errors"
+	"golang.org/x/crypto/chacha20poly1305"
+	"os"
+	"time"
+)
+
+// Simple helper function to convert config int to time.Duration
+func helperSecondsToDuration(seconds int) time.Duration {
+	return time.Duration(seconds) * time.Second
+}
+
+// Simple helper function to check file exists
+func exists(path string, checkDirectory bool) (bool, error) {
+	info, err := os.Stat(path)
+	if !checkDirectory && err == nil && !info.IsDir() {
+		return true, nil
+	}
+	if checkDirectory && err == nil && info.IsDir() {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+// Simple helper function to check file is readable
+func fileExistsAndIsReadable(path string) error {
+	exists, err := exists(path, false)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errors.New("file does not exist")
+	}
+	fil, err := os.OpenFile(path, os.O_RDONLY, 0)
+	if err != nil {
+		return err
+	}
+	fil.Close()
+	return nil
+}
+
+// Simple encryption helper
+func helperEncrypt(key []byte, plaintext []byte) ([]byte, error) {
```

Comment:
I cannot say I am charmed by undocumented DIY crypto implementations.

Also please read the points in https://github.com/PowerDNS/simpleblob/issues/52#issuecomment-1746057711

Perhaps any client side encryption should be implemented as an swappable middleware layer between simpleblob and the application that can be used by multiple backends?

Reply:
On the subject of "undocumented DIY crypto implementations."

Far from the case. I would point out that it is (a) nothing but a thin-wrapper around Go stdlib crypto (b) the implementation is basically nothing more than a copy/paste of the example from the Go docs (https://pkg.go.dev/golang.org/x/crypto/chacha20poly1305#example-NewX).   The only significant difference is I don't `panic()` but return the error.  :wink:

Thanks for all the feedback though, much appreciated food for thought.